### PR TITLE
Feature/android toggle camera improvement

### PR DESCRIPTION
**Pull request by himanshut0305**
Added Camera toggle feature for android in broadcast event, added proper intent builder for the same in android SDK,
this gives an additional control to the integrator to manage the camera facing mode with the help of event emitted.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -86,6 +86,7 @@ dependencies {
         // Firebase
         //  - Crashlytics
         //  - Dynamic Links
+
```
Comment (saghul) [inline android/app/build.gradle:89]: Unnecessary change, please undo it.

File: android/app/src/main/java/org/jitsi/meet/MainActivity.java [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 import androidx.annotation.Nullable;
 
+
```
Comment (saghul) [inline android/app/src/main/java/org/jitsi/meet/MainActivity.java:33]: ditto

File: android/app/src/main/java/org/jitsi/meet/MainActivity.java [hunk 3]
```diff
@@ -87,7 +88,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     protected boolean extraInitialize() {
-        Log.d(this.getClass().getSimpleName(), "LIBRE_BUILD="+BuildConfig.LIBRE_BUILD);
+        Log.d(this.getClass().getSimpleName(), "LIBRE_BUILD="+ BuildConfig.LIBRE_BUILD);
```
Comment (saghul) [inline android/app/src/main/java/org/jitsi/meet/MainActivity.java:91]: ditto

File: android/app/src/main/java/org/jitsi/meet/MainActivity.java [hunk 4]
```diff
@@ -173,7 +174,7 @@ private void resolveRestrictions() {
                     restrictions.containsKey(RESTRICTION_SERVER_URL)) {
                     defaultURL = restrictions.getString(RESTRICTION_SERVER_URL);
                     configurationByRestrictions = true;
-                // Otherwise use default URL from app-restrictions.xml.
+                    // Otherwise use default URL from app-restrictions.xml.
```
Comment (saghul) [inline android/app/src/main/java/org/jitsi/meet/MainActivity.java:177]: ditto

File: android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java [hunk 5]
```diff
@@ -8,6 +8,10 @@ public static Intent buildSetAudioMutedIntent(boolean muted) {
         intent.putExtra("muted", muted);
         return intent;
     }
+    public static Intent buildToggleCameraIntent() {
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java:11]: please leave an empty line between functions

File: android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java [hunk 6]
```diff
@@ -8,6 +8,10 @@ public static Intent buildSetAudioMutedIntent(boolean muted) {
...
+        Intent intent = new Intent(BroadcastAction.Type.TOGGLE_CAMERA_FACING_MODE.getAction());
```
Comment (saghul) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java:12]: No need to create a variable just for returning it immediately

File: react/features/mobile/external-api/middleware.js [hunk 7]
```diff
@@ -27,7 +27,8 @@ import {
 } from '../../base/connection';
 import { JitsiConferenceEvents } from '../../base/lib-jitsi-meet';
 import { MEDIA_TYPE } from '../../base/media';
-import { SET_AUDIO_MUTED, SET_VIDEO_MUTED } from '../../base/media/actionTypes';
+// ,TOGGLE_CAMERA_FACING_MODE
```
Comment (saghul) [inline react/features/mobile/external-api/middleware.js:30]: ?

File: react/features/mobile/external-api/middleware.js [hunk 8]
```diff
@@ -286,6 +288,10 @@ function _registerForNativeEvents(store) {
     eventEmitter.addListener(ExternalAPI.SET_VIDEO_MUTED, ({ muted }) => {
         dispatch(muteLocal(muted, MEDIA_TYPE.VIDEO));
     });
+    /*Added listner for event toggle camera for android*/
```
Comment (saghul) [inline react/features/mobile/external-api/middleware.js:291]: drop the comment

**Review by saghul — CHANGES_REQUESTED** [review]

**Comment by saghul** [comment]
I'd like to see this implemented for iOS.

File: android/app/build.gradle (hunk 1 above)
Comment (himanshut0305) [reply android/app/build.gradle]: done, thanks for assist!

File: android/app/src/main/java/org/jitsi/meet/MainActivity.java (hunk 2 above)
Comment (himanshut0305) [reply android/app/src/main/java/org/jitsi/meet/MainActivity.java]: ditto

File: android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java (hunk 5 above)
Comment (himanshut0305) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/BroadcastIntentHelper.java:12]: done!

**Review by himanshut0305 — COMMENTED** [review]
made all the changes requested.

File: package.json [hunk 9]
```diff
@@ -48,10 +48,10 @@
     "i18next": "17.0.6",
     "i18next-browser-languagedetector": "3.0.1",
     "i18next-xhr-backend": "3.0.0",
-    "jQuery-Impromptu": "github:trentrichardson/jQuery-Impromptu#v6.0.0",
     "jitsi-meet-logger": "github:jitsi/jitsi-meet-logger#v1.0.0",
     "jquery": "3.5.1",
     "jquery-i18next": "1.2.1",
+    "jQuery-Impromptu": "github:trentrichardson/jQuery-Impromptu#v6.0.0",
```
Comment (vp8x8) [inline package.json:54]: We don't use this addon anymore.

**Pull request closed without merging**